ludwig/features/base_feature.py

```diff
@@ -380,9 +379,18 @@ def get_metrics(self):
         metric_vals = {}
         for metric_name, metric_fn in self._metric_functions.items():
             try:
-                metric_vals[metric_name] = get_scalar_from_ludwig_metric(metric_fn)
-            except Exception:
-                logger.exception(f"Caught exception computing metric: {metric_name}.")
+                computed_metric = metric_fn.compute()
+            except Exception as e:
+                logger.exception(f"Caught exception computing metric: {metric_name} with error: {e}.")
+                continue
+
+            # Metrics from torchmetrics can be a tensor.
+            if isinstance(computed_metric, Tensor):
+                metric_vals[metric_name] = computed_metric.detach().cpu().numpy().item()
+            else:
+                # Or a dict of tensors. Unpack if so.
+                for sub_metric_name, metric in computed_metric.items():
+                    metric_vals[sub_metric_name] = metric.detach().cpu().numpy().item()
```

Comment:
Can we leave a comment here saying what kinds of metrics may be a dictionary (rouge)?

Reply:
Done.